src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -251,7 +251,8 @@ def _cancel_job(self, volname, job):
         canceled = False
         log.info("canceling job {0} for volume {1}".format(job, volname))
         try:
-            if not volname in self.q and not volname in self.jobs and not job in self.jobs[volname]:
+            vol_jobs = [j[0] for j in self.jobs[volname]]
```

Comment:
`self.jobs.get(volname, [])` to handle missing `volname` in `self.jobs`?

Then the below conditional can be simplified too.

Reply:
Done